packages/app/src/systems/Core/utils/string.tsx

```diff
@@ -21,3 +21,41 @@ export const truncate = (str: string, length: number = 30) => {
   }
   return str;
 };
+
+export type GetUniqueStringProps = {
+  desired?: string;
+  allValues: Array<string | undefined>;
+};
+
+// get unique string adding suffix to desired value if it already exists in allValues
+export const getUniqueString = ({
+  desired,
+  allValues,
+}: GetUniqueStringProps): string | undefined => {
+  if (!desired) return desired;
+
+  function nextNotRepeated(opt: { value: string; tries?: number }) {
+    const { value, tries = 1 } = opt;
+
+    const repeatedAssets = allValues.filter(
+      (a) => a && a.trim() === value.trim()
+    );
+
+    let notRepeatedField: string;
+    if (repeatedAssets[0]) {
+      const nextToTry =
+        tries === 1
+          ? `${value} (${tries})`
+          : `${value?.slice(0, -4)} (${tries})`;
```

Comment:
Why we need this slice here?

Reply:
remove last (1) and inform new (2) for example.

Reply:
And if the number is `Text (10)` What happens?

Reply:
good catch my friend